code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -371,3 +371,38 @@
 	desc = "An upgraded model of the basic laser gun. There seems to be some sort of slot in the handle."
 	can_take_pai = TRUE
 	origin_tech = Tc_COMBAT + "=3;" + Tc_MAGNETS + "=2;" + Tc_ENGINEERING + "=2;" + Tc_PROGRAMMING + "=4"
+
+/obj/item/weapon/gun/energy/laser/rainbow
+
+	name = "rainbow laser"
+	desc = "A fearsome gun used by clown special forces"
+	projectile_type = "/obj/item/projectile/beam/rainbow"
+	var/current_color = 1
+	var/list/color_list = list("#FF0000","#FF8C00","#FFFF00","#00FF00","#00BFFF","#0000FF","#9400D3")
+	var/fire_mode = 1
+
+/obj/item/weapon/gun/energy/laser/rainbow/attack_self(mob/living/user as mob)
```

Comment:
The proc is used as a verb it communicates to BYOND what type of input is valid for the arguments, for the built in input windows.

This however has no relevance when working without regular procs so it's only clutter, as the type is better communicated by casting the variable.